internal/handlers/restore_backup.go

```diff
@@ -0,0 +1,165 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+	"github.com/ydb-platform/ydb-go-genproto/protos/Ydb"
+	"github.com/ydb-platform/ydb-go-genproto/protos/Ydb_Operations"
+	"go.uber.org/zap"
+	"time"
+	"ydbcp/internal/config"
+	"ydbcp/internal/connectors/client"
+	"ydbcp/internal/connectors/db"
+	"ydbcp/internal/types"
+	"ydbcp/internal/util/xlog"
+)
+
+func MakeRBOperationHandler(db db.DBConnector, client client.ClientConnector, config config.Config) types.OperationHandler {
+	return func(ctx context.Context, op types.Operation) error {
+		return RBOperationHandler(ctx, op, db, client, config)
+	}
+}
+
+func isValidOperationResponse(resp *Ydb_Operations.GetOperationResponse) bool {
+	return resp.GetOperation().GetStatus() == Ydb.StatusIds_SUCCESS ||
+		resp.GetOperation().GetStatus() == Ydb.StatusIds_CANCELLED
+}
+
+func RBOperationHandler(
+	ctx context.Context,
+	operation types.Operation,
+	db db.DBConnector,
+	client client.ClientConnector,
+	config config.Config,
+) error {
+	xlog.Info(ctx, "received operation",
+		zap.String("id", operation.GetId().String()),
+		zap.String("type", string(operation.GetType())),
+		zap.String("state", string(operation.GetState())),
+		zap.String("message", operation.GetMessage()),
+	)
+
+	if operation.GetType() != types.OperationTypeRB {
+		return fmt.Errorf("wrong operation type %s != %s", operation.GetType(), types.OperationTypeRB)
+	}
+
+	mr, ok := operation.(*types.RestoreBackupOperation)
+	if !ok {
+		return fmt.Errorf("can't cast Operation to RestoreBackupOperation %s", types.OperationToString(operation))
+	}
+
+	conn, err := client.Open(ctx, types.MakeYdbConnectionString(mr.YdbConnectionParams))
+	if err != nil {
+		return fmt.Errorf("error initializing client connector %w", err)
+	}
+
+	defer func() { _ = client.Close(ctx, conn) }()
+
+	opResponse, err := client.GetOperationStatus(ctx, conn, mr.YdbOperationId)
+	if err != nil {
+		if (mr.CreatedAt.Unix() + config.OperationTtlSeconds) <= time.Now().Unix() {
+			operation.SetState(types.OperationStateError)
+			operation.SetMessage("Operation deadline exceeded")
+			return db.UpdateOperation(ctx, operation)
+		}
+
+		return fmt.Errorf(
+			"failed to get operation status for operation id %s, import operation id %s: %w",
+			mr.GetId().String(),
+			mr.YdbOperationId,
+			err,
+		)
+	}
+
+	if !isValidOperationResponse(opResponse) {
```

Comment:
Do we want to set Error if it is UNAVAILABLE or OVERLOADED? It is hard to say if we should Error or skip